pkg/repro/repro.go

```diff
@@ -151,11 +151,44 @@ func Run(crashLog []byte, cfg *mgrconfig.Config, reporter report.Reporter, vmPoo
 		return nil, err
 	}
 	if res != nil {
-		ctx.reproLog(3, "repro crashed as:\n%s", string(ctx.report))
-		res.Stats = ctx.stats
+		// Save current report.
 		res.Title = ctx.title
 		res.Log = ctx.log
 		res.Report = ctx.report
+		ctx.reproLog(3, "repro crashed as:\n%s", string(ctx.report))
+
+		// Try to rerun the repro if the report is corrupted.
+		rep := reporter.Parse(ctx.log)
```

Comment:
We've just parsed the report in testImpl. I think we better remember corrupted flag along with title/log/report (perhaps, remember whole `*report.Report`).

Reply:
Done.